apps/ledger-live-mobile/src/components/CustomImage/imageUtils.ts

```diff
@@ -0,0 +1,144 @@
+    if (result.cancelled) {
+      return null;
+    }
+    const { uri, width, height } = result;
+    if (uri) {
+      return {
+        width,
+        height,
+        imageFileUri: uri,
+      };
+    }
+    throw new Error("uri is falsy");
+  } catch (e) {
+    console.error(e);
+    throw new ImageLoadFromGalleryError();
+  }
+}
+
+export function downloadImageToFileWithCancellable({
+  imageUrl,
+}: ImageUrl): [Promise<ImageFileUri>, StatefulPromise<FetchBlobResponse>] {
+  const downloadTask = RNFetchBlob.config({ fileCache: true }).fetch(
+    "GET",
+    imageUrl,
+  );
+  return [
+    downloadTask
+      .then(res => ({ imageFileUri: "file://" + res.path() }))
+      .catch(() => {
+        throw new ImageDownloadError();
+      }),
+    downloadTask,
+  ];
+}
+
+export async function downloadImageToFile({
+  imageUrl,
+}: ImageUrl): Promise<ImageFileUri> {
+  return downloadImageToFileWithCancellable({ imageUrl })[0];
+}
+
+export async function loadImageToFileWithDimensions(
+  source: Partial<ImageUrl> & Partial<ImageFileUri> & Partial<ImageDimensions>,
+): Promise<ImageFileUri & Partial<ImageDimensions>> {
+  if (source?.imageFileUri) {
+    return {
+      imageFileUri: source?.imageFileUri,
+      height: source?.height,
+      width: source?.width,
+    };
+  }
+  if (source?.imageUrl) {
+    const { imageUrl } = source;
+    const [downloadPromise, downloadTask] = downloadImageToFileWithCancellable({
+      imageUrl,
+    });
+    try {
+      const [dims, { imageFileUri }] = await Promise.all([
+        loadImageSizeAsync(imageUrl),
+        downloadPromise,
+      ]);
+      return {
+        width: dims.width,
+        height: dims.height,
+        imageFileUri,
+      };
+    } catch (e) {
+      /**
+       * in case an error happens in loadImageSizeAsync we have to cancel the
+       * download task
+       * */
+      downloadTask.cancel();
+      throw e;
+    }
+  }
+  throw new Error("No source specified"); // this shouldn't happen
+}
+
+export async function loadImageSizeAsync(
+  url: string,
+): Promise<ImageDimensions> {
+  return new Promise((resolve, reject) => {
+    Image.getSize(
+      url,
+      (width, height) => {
+        resolve({ width, height });
+      },
+      error => {
+        console.error(error);
+        if (error?.message?.startsWith("Pool hard cap violation? ")) {
+          reject(new ImageTooLargeError());
+        } else {
+          reject(new ImageMetadataLoadingError());
+        }
+      },
+    );
+  });
+}
+
+export function fitImageContain(
+  imageDimensions: ImageDimensions,
+  boxDimensions: ImageDimensions,
+): { height: number; width: number } {
+  const { height: imageHeight, width: imageWidth } = imageDimensions;
+  const { height: boxHeight, width: boxWidth } = boxDimensions;
+  if (imageHeight < boxHeight && imageWidth < boxWidth)
+    return {
+      width: imageWidth,
+      height: imageHeight,
+    };
+  if (imageWidth / imageHeight >= boxWidth / boxHeight) {
```

Comment:
i understand the idea of the ratio, but to guess if the width or the height is the constraint, why not just check `imageWidth > boxWidth` for the width is the constraint (i might miss something ?)

And if `imageHeight === 0` or `boxHeight === 0` the division will fail 🙈

Reply:
:beaten_trollface:

Reply:
So indeed `0` values with fail haha I will fix that.

Reply:
Regarding the logic itself, just checking `imageWidth > boxWidth` is not enough, both the width and the height can be the constraint.

The question is which dimension (width or height) is the most constraining.

If it's the width (this case), then we use the width of the box as a basis for resizing (while keeping the original aspect ratio of the image) and then we are sure that the resulting height will fit in the box.

Reply:
@alexandremgo I added some comments in the code to explicit what is happening.

Reply:
Love the explanation, thanks :)